### ISSUE TITLE
Queries aren't returning any records

### ISSUE THREAD

**Issue by NathanWailes** [mitsuhiko/flask-sqlalchemy#562]
I'm having a problem with my local version of my website. I'm using Flask, Flask-SQLAlchemy, and Python 3.5.2, and calls to my MySQL database that had worked in the past (although on a different laptop) are now not working, instead returning the error "MySQL Connection not available".

Somewhat bizarrely, the exact behavior I'm seeing is that when I try to log into my website the first time, the database call returns "None" for the user, even though the user is present in the database. When I then reload the login page and try again, I get sent to an error page that says "MySQL Connection not available".

I have tried running a debugger and running a query that returns all of the database records, and it returns an empty list, even though there should be a single record (my test account).

I have forum posts of others who have had this issue, but those fixes have not worked.

- I already had `SQLALCHEMY_POOL_RECYCLE` set to a lower value than the corresponding MySQL variables (as recommended [here](https://stackoverflow.com/a/26898700/4115031)).
- I had the `SQLALCHEMY_POOL_RECYCLE` value set to 299, as recommended [here](https://blog.pythonanywhere.com/121/).
- Changing my MySQL `wait_timeout` and `interactive_timeout` values to be much lower (300 from something like 28800), as described [here](https://stackoverflow.com/a/32211825/4115031), didn't seem to have any effect.
- Putting my queries within a `try` / `except` with a `db.session.rollback()` in the `except` clause (as described [here](https://stackoverflow.com/a/32211825/4115031) and rec'd [here](https://www.pythonanywhere.com/forums/topic/885/#id_post_6354)) does stop the app from crashing, but it still says it isn't finding the record that is clearly there.

It seems like there may be two problems here:

- The query isn't finding the record I know is there.
- When I try to run the query a second time, it returns the "MySQL Connection not available" error.

Other things tried:

- If I run the debugger and try inserting records into the database, they show up in my PyCharm database viewer.
- Adding the line `db.init_app(app)` to the bottom of my app.py initialization function stops the error messages from crashing the app, but still doesn't result in the ORM finding the user record.

**Comment by davidism** [comment]
I understand MySQL behavior is frustrating. If possible, try a different database such as PostgreSQL and see if the issue persists. Either way, please use Stack Overflow for questions about your own code. Be sure to include a [minimal, complete, and verifable example](https://stackoverflow.com/help/mcve).

**Comment by NathanWailes** [comment]
Alright, well, I finally got it working.

For anyone else who maybe runs into this error, here's what worked for me:

I had recently added a new column to a table, but I somehow ended up with the name of the column as "uuid", and a default of an empty string (it was a String field). I fixed those two things, dropped the schema and re-created all the tables, and everything is working now. I still don't understand why things weren't working, but at least the issue is fixed now.
